scripts/build-targz.sh

```diff
@@ -29,17 +29,13 @@ build-targz () {
         setup.cfg \
         setup.py \
         ulauncher \
-        ulauncher.desktop.in \
+        ulauncher.desktop \
         $tmpdir \
         --exclude-from=.gitignore
 
     # This is only needed because data/preferences is in .gitignore
     cp -r data/preferences $tmpdir/data/preferences
 
-    # set version to a tag name ($1)
-    sed -i "s/__version__ =.*/__version__ = '$1'/g" $tmpdir/ulauncher/config.py
```

Comment:
So these build scripts will no longer accept the version as an argument to the script? Though the filename below does still accept the argument in order to construct the filename. What does it mean if the value given to this script doesn't match the value hardcoded in the new VERSION file?

Reply:
The build scripts work exactly the same and takes the version as an argument for the file name. The line removed was just for handling this: https://github.com/friday/Ulauncher-Revived/blob/dev/ulauncher/config.py#L9, which is now handled here instead: https://github.com/friday/Ulauncher-Revived/blob/5704714038de88346cff08a1b7e7512c6b72411e/setup.py#L31

I think we don't really need the targz-script btw. `./setup.py bdist --formats=gztar` does the same.

Reply:
The results of `setup.py bdist --formats=gztar` are very different though. The existing scripts generate a source tarball which can be used along with setup to install in the standard ways. Possibly in the system location, or the user location, or used as the starting point for packaging (like in the [main fedora repos](https://src.fedoraproject.org/rpms/ulauncher/blob/rawhide/f/ulauncher.spec#_57)). The output of bdist assumes you just want to lay the tarball down at your root filesystem level, something like `tar xf ulauncher.tar.gz -C /`. I'm not sure that is the expected type of tarball to deliver.

Reply:
> The results of `setup.py bdist --formats=gztar` are very different though...

I know. This is all true. I think the differences are actually small, but meaningful. I thought I could work around this enough to use bdist, plus basically the `tar xf ulauncher.tar.gz -C /` approach. But that means the system paths have to be the same on the release side (us) as the end user side, or that the release side has to adjust to use standard paths and the end user side has to make the adjustments. On Arch Linux  Ulauncher ends up in `/usr/lib/python3.9/site-packages/ulauncher/` for example, which wouldn't work with most other distros, and not be optimal (but still work) for later versions of Arch, so let's stick to the custom targz script or a version of it. :+1:

Reply:
on this point, I think the closer analog to the existing tarball release is probably `setup.py sdist`, though that doesn't include the "preprocessed" preferences that the current tarball does. If there are similar hooks for sdist as for build, then it could conceivably entirely reproduce a similar source release.